packages/amplication-git-utils/src/git/github.service.ts

```diff
@@ -1,82 +1,209 @@
-import { Injectable } from "@nestjs/common";
-import { ConfigService } from "@nestjs/config";
 import { createAppAuth } from "@octokit/auth-app";
 import { components } from "@octokit/openapi-types";
 import { App, Octokit } from "octokit";
-import { createPullRequest } from "octokit-plugin-create-pull-request";
-import { Changes } from "octokit-plugin-create-pull-request/dist-types/types";
 import {
   EnumGitOrganizationType,
   EnumPullRequestMode,
   GithubFile,
+  File,
   RemoteGitOrganization,
   RemoteGitRepos,
   RemoteGitRepository,
+  GitProviderArgs,
+  GetRepositoryArgs,
+  GetRepositoriesArgs,
+  CreateRepositoryArgs,
+  CreatePullRequestArgs,
 } from "../types";
 import { ConverterUtil } from "../utils/convert-to-number";
 import { UNSUPPORTED_GIT_ORGANIZATION_TYPE } from "./git.constants";
 import { AccumulativePullRequest } from "./github/AccumulativePullRequest";
 import { BasicPullRequest } from "./github/BasicPullRequest";
+import { createPullRequest } from "octokit-plugin-create-pull-request";
+import { Changes } from "octokit-plugin-create-pull-request/dist-types/types";
 
 const GITHUB_FILE_TYPE = "file";
-export const GITHUB_CLIENT_SECRET_VAR = "GITHUB_CLIENT_SECRET";
-export const GITHUB_APP_APP_ID_VAR = "GITHUB_APP_APP_ID";
-export const GITHUB_APP_PRIVATE_KEY_VAR = "GITHUB_APP_PRIVATE_KEY";
-export const GITHUB_APP_INSTALLATION_URL_VAR = "GITHUB_APP_INSTALLATION_URL";
-export const UNEXPECTED_FILE_TYPE_OR_ENCODING = `Unexpected file type or encoding received`;
 
 type DirectoryItem = components["schemas"]["content-directory"][number];
-@Injectable()
+
 export class GithubService {
   private app: App;
+  private appId: string;
+  private privateKey: string;
   private gitInstallationUrl: string;
 
-  constructor(private readonly configService: ConfigService) {
-    this.gitInstallationUrl = this.configService.get(
-      GITHUB_APP_INSTALLATION_URL_VAR
-    );
+  constructor(private readonly gitProviderArgs: GitProviderArgs) {
+    this.init();
+  }
 
-    const appId = this.configService.get(GITHUB_APP_APP_ID_VAR);
-    const privateKey = this.configService
-      .get(GITHUB_APP_PRIVATE_KEY_VAR)
-      .replace(/\\n/g, "\n");
+  init(): void {
+    const {
+      GITHUB_APP_INSTALLATION_URL,
+      GITHUB_APP_APP_ID,
+      GITHUB_APP_PRIVATE_KEY,
+    } = process.env;
```

Comment:
Libraries must not use environment variables unless clearly documented in the library documentation.